### ISSUE TITLE
out of tree build fails: "'_DefaultIsolatedEnv' object has no attribute '_scripts_dir'"

### ISSUE THREAD

**Issue by kaktus42** [pyodide/pyodide#4655]
## 🐛 Bug

I am trying to do an "out of tree build" of the package [aioesphomeapi](aioesphomeapi).
I followed the procedure outlined [in the docs](https://pyodide.org/en/stable/development/building-and-testing-packages.html#building-and-testing-packages-out-of-tree).
At the step to build (`pyodide build`) I get the following error:

```
✗ pyodide build                                                    
xbuild environment already exists, skipping download                                                                                                                                                                                                                           
Installing xbuild environment                                                                                                                                                                                                                                                  

Traceback (most recent call last):
  File "/home/kaktus42/miniconda3/lib/python3.10/site-packages/pyodide_build/pypabuild.py", line 71, in _runner
    env["BUILD_ENV_SCRIPTS_DIR"] = isolated_build_env._scripts_dir
AttributeError: '_DefaultIsolatedEnv' object has no attribute '_scripts_dir'

ERROR Backend operation failed: AttributeError("'_DefaultIsolatedEnv' object has no attribute '_scripts_dir'")
```

The "normal" build (`python -m build`) works without problems.


### To Reproduce

Just running the build as described.

### Expected behavior

Successful build.


### Environment

- Pyodide Version: Pyodide CLI Version: 0.2.3
- Browser version: -
- Any other relevant information: 

pyodide-build==0.25.0
pyodide-cli==0.2.3
pyodide-lock==0.1.0a4

```
✗ ./emsdk activate ${PYODIDE_EMSCRIPTEN_VERSION}
Resolving SDK version '3.1.46' to 'sdk-releases-21644188d5c473e92f1d7df2f9f60c758a78a486-64bit'
Setting the following tools as active:
   node-16.20.0-64bit
   releases-21644188d5c473e92f1d7df2f9f60c758a78a486-64bit
```


### Additional context

-

**Comment by ryanking13** [comment]
This will be fixed by #4653 (we will release pyodide-build 0.25.1 soon with that patch). For pyodide-build 0.25.0, downgrading the build version:

```
pip install build<1.2.0
```

will resolve the issue.

**Comment by ryanking13** [comment]
Closing as resolved by #4653
